[fix][doc] Workaround Go Yaml issue go-yaml/yaml#789 in docker-compose example

lhotari opened the pull request:
Fixes #24039

### Motivation

docker-compose examples don't work due to go-yaml/yaml#789 issue. 
It's necessary to replace `command: >` with `command: |`. More details in #24039 comments.
Similar change was made to website docs in https://github.com/apache/pulsar-site/commit/0df29a62.

### Modifications

- replace `command: >` with `command: |` in the docker-compose example
- remove Pulsar Manager component
- remove Pulsar SQL component
- adjust max heap size parameters to run the cluster with reasonable resources

### Documentation

<!-- DO NOT REMOVE THIS SECTION. CHECK THE PROPER BOX ONLY. -->

- [ ] `doc` <!-- Your PR contains doc changes. -->
- [ ] `doc-required` <!-- Your PR changes impact docs and you will update later -->
- [x] `doc-not-needed` <!-- Your PR changes do not impact docs -->
- [ ] `doc-complete` <!-- Docs have been already added -->

lhotari commented:
/pulsarbot rerun-failure-checks

visortelle reviewed (commented):
The changes LGTM, but the SQL worker container restarts in a loop:

```
# sql1
# Restarting (1) 42 seconds ago  
...
sql1         | 
sql1         | -- Invalid command 'sql-worker' -- Use '/pulsar/bin/pulsar help' to get a list of valid commands
sql1         | 
sql1 exited with code 1
```

Pulsar SQL was removed a while ago: https://github.com/apache/pulsar/discussions/23435
Maybe we should also remove this container in docker-compose.yml?

Also, we probably can remove the version attribute because of this warning: 

```
WARN[0000] /home/xxxxxxxxxxxxxxxx/apache/pulsar/docker-compose/kitchen-sink/docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion 
```

lhotari commented:
> The changes LGTM, but the SQL worker container restarts in a loop:

Thanks, @visortelle . I removed Pulsar SQL and Pulsar Manager from the example and adjusted the max heap size settings so that it will startup with reasonable resources.
